client/src/components/TodoApp/TodoApp.jsx

```diff
@@ -0,0 +1,150 @@
+import React, { useEffect, useState } from "react";
+import styles from "./TodoApp.module.css";
+import MyToast from "../MyToast/MyToast";
+import {
+  handleNewItem,
+  getAllTasks,
+  deleteAllTasks,
+} from "../../serverApi/itemClient";
+import ListItems from "../ListItems/ListItems";
+import UserInput from "../UserInput/UserInput";
+import Sort from "../Sort/Sort";
+import Button from "../Button/Button";
+import { checkIfInputIsNotValid } from "../../utils/utils";
+
+const TodoApp = () => {
+  const [data, setData] = useState([]);
```

Comment:
if your data was a dictionary object and not an array you'll had easier time handling it.

your delete below won't need a mapping/filter just delete data[id]
your update won't need mapping just a setData(...data, {<updated/new item>})
etc.